controllers/controllers.go

```diff
@@ -177,7 +177,14 @@ func windowsNodePredicate(byoh bool) predicate.Funcs {
 			return false
 		},
 		DeleteFunc: func(e event.DeleteEvent) bool {
-			return false
+			if e.Object.GetLabels()[core.LabelOSStable] != "windows" {
+				return false
+			}
+			if (byoh && e.Object.GetLabels()[BYOHLabel] != "true") ||
+				(!byoh && e.Object.GetLabels()[BYOHLabel] == "true") {
+				return false
+			}
```

Comment:
Please add a helper as we are just copy pasting the same code in multiple places.